### ISSUE TITLE
Flow dev reset state

### ISSUE THREAD

**Issue by sideninja** [onflow/flow-cli#770]
During development using the flow dev command constantly removes and updates contracts which is good for flexibility on the changes and restrictions of the update. The problem is that if there is an existing state established on the account that might no longer be valid after the update. I think it would be better if we first take a snapshot of the state on the emulator and then rollback to that and deploy contracts, that way we are sure to start with a fresh state. 

Blocked by: https://github.com/onflow/flow-emulator/issues/309

**Comment by bjartek** [comment]
This would be a really cool feature. Taking a state snapshot and reapplying it would also be very nice to have for integration tests. You can set up your state the way you want it. Store it and start the test from that known state.

**Comment by sideninja** [comment]
Yep. I think this can be the first improvement of the flow dev. I want to launch in the current shape tho because this is a bit more technical and it would take some time to finish.

**Comment by sideninja** [comment]
Related to work on emulator for rollback of blocks

**Comment by bjartek** [comment]
I belive @bluesign is working on something that will enable this.

**Comment by bluesign** [comment]
snapshots are already active ( today merged ) also I did a PR for rollback today: https://github.com/onflow/flow-emulator/pull/341

**Comment by bjartek** [comment]
amazing

**Comment by sideninja** [comment]
Closing due to: https://github.com/onflow/flow-cli/issues/986
